### PR TITLE
if there are no external services, redirect to external-services-add

### PR DESCRIPTION
**Pull request by beyang**
During a recent user onboarding, @slimsag and I witnessed user confusion that occurred when a new user wanted to add repositories to Sourcegraph, navigated to the external services page, but didn't understand what to do (click the blue button to "add a new external service"). We should just redirect to the "add external service" page if there are no existing external services.

image

should redirect to

image

### CODE REVIEW

File: web/src/site-admin/SiteAdminExternalServicesPage.tsx [hunk 1]
```diff
@@ -122,11 +127,26 @@ class FilteredExternalServiceConnection extends FilteredConnection<
...
+
+    constructor(props: Props) {
+        super(props)
+        this.state = {}
+    }
 
     public componentDidMount(): void {
         eventLogger.logViewEvent('SiteAdminExternalServices')
+        this.subscriptions.add(
+            this.queryExternalServices({ first: 1 })
```
Comment (hadrian-git) [inline web/src/site-admin/SiteAdminExternalServicesPage.tsx]: nit:
```suggestion
            this.queryExternalServices({ first: 1 }).subscribe(externalServicesResult =>
                this.setState({ noExternalServices: externalServicesResult.totalCount === 0 })
            )
```

**Comment by slimsag** [comment]
Note I actually filed this exact issue back in May of last year: https://github.com/sourcegraph/sourcegraph/issues/4077

**Comment by beyang** [comment]
Fix https://github.com/sourcegraph/sourcegraph/issues/4077

File: web/src/site-admin/SiteAdminExternalServicesPage.tsx (hunk 1 above)
Comment (beyang) [reply web/src/site-admin/SiteAdminExternalServicesPage.tsx]: Will fix in a follow-up PR.

**Pull request merged**